### PR TITLE
Api change to let run set_attr and get_attr remotely

### PR DESCRIPTION
**Pull request by mariostieriansys**
The API now runs set_attr and get_attr directly in EnSight, without looping over the ensobjlist on the PyEnSight side 

I computed some performance against a 10k part synthetic dataset:

new branch

same session

session.ensight.objs.core.PARTS.set_attr("SELECTED", True): 263.00 s
session.ensight.objs.core.PARTS.get_attr("VISIBLE", True): 3.70 s

different sessions

session.ensight.objs.core.PARTS.set_attr("SELECTED", True): 243.37 s
session.ensight.objs.core.PARTS.get_attr("VISIBLE", True): 255.68

Main

same session

session.ensight.objs.core.PARTS.set_attr("SELECTED", True): 347.32 s
session.ensight.objs.core.PARTS.get_attr("VISIBLE", True): 82.43 s

different sessions

session.ensight.objs.core.PARTS.set_attr("SELECTED", True): 344.86 s
session.ensight.objs.core.PARTS.get_attr("VISIBLE", True): 344.98 s

So, apart from an overall 30% performance improvement, you also get a huge boost on following operations on the same list. This is because the code is now building an ensobjlist on the EnSight side. So, in the following operations, EnSight will create much faster the same list, using the cached data from the previous computation

### CODE REVIEW

**Comment by kecolburn** [comment]
I appreciate being included in the review. However, I don't know enough about the internal implementation and execution of this logic to make any sensible critique.

**Review by kecolburn — APPROVED** [review]

File: src/ansys/pyensight/core/listobj.py [hunk 1]
```diff
@@ -141,13 +141,20 @@ def set_attr(self, attr: Any, value: Any) -> int:
...
         count: int = 0
+        objid_list = []
+        session = None
         for item in self:
             if isinstance(item, ENSOBJ):
                 try:
-                    item.setattr(attr, value)
+                    if not session:
+                        session = item._session
+                    objid_list.append(item.__OBJID__)
```
Comment (margalva) [inline src/ansys/pyensight/core/listobj.py:151]: This works, but I feel that this for item loop would be faster if you did it like:

objid_list = [x.__OBJID__ for x in self if isinstance(x, ENSOBJ)]
count = len(objid_list)

for i in objid_list:
    if not session:
         session = item._session
    else:
        break
        
        
I don't know if large item lists are a concern or not for you, so up to you
@mariostieriansys

**Review by margalva — APPROVED** [review]
Approved with a suggestion (that applies to both methods). Up to you to take it or leave it.

File: src/ansys/pyensight/core/listobj.py (hunk 1 above)
Comment (mariostieriansys) [reply src/ansys/pyensight/core/listobj.py:151]: @margalva thanks for the suggestion, you're right with the list comprehension I can make it faster

File: src/ansys/pyensight/core/listobj.py [hunk 2]
```diff
@@ -141,13 +141,17 @@ def set_attr(self, attr: Any, value: Any) -> int:
...
-            if isinstance(item, ENSOBJ):
-                try:
-                    item.setattr(attr, value)
-                    count += 1
-                except RuntimeError:
-                    pass
+            if hasattr(item, "_session"):
+                session = item._session
+                break
+        if session:
```
Comment (randallfrank) [inline src/ansys/pyensight/core/listobj.py:152]: @mariostieriansys The return value of set_attrs is not the number of items in the input list, rather the number of items in the input list that could successfully have the values set.   I believe the return value should be:

count = session.cmd(msg)

Consider a mixed list of annots where some support the attribute and some do not.

**Review by randallfrank — CHANGES_REQUESTED** [review]

File: src/ansys/pyensight/core/listobj.py (hunk 2 above)
Comment (mariostieriansys) [reply src/ansys/pyensight/core/listobj.py:152]: makes sense, I am fixing it

File: src/ansys/pyensight/core/session.py [hunk 3]
```diff
@@ -972,6 +972,7 @@ def load_data(
         reader_options: Optional[dict] = None,
         new_case: bool = False,
         representation: str = "3D_feature_2D_full",
+        monitor_new_steps: bool = False,
```
Comment (randallfrank) [inline src/ansys/pyensight/core/session.py:975]: Not in the docstrings?

**Review by randallfrank — DISMISSED** [review]

File: src/ansys/pyensight/core/session.py (hunk 3 above)
Comment (mariostieriansys) [reply src/ansys/pyensight/core/session.py:975]: I don't know why it's there. I am fixing it

Comment (randallfrank) [reply src/ansys/pyensight/core/session.py:975]: I would not doc it in this PR as you are changing the API a bit in the next PR.

Comment (mariostieriansys) [reply src/ansys/pyensight/core/session.py]: it shouldn't have been there at all, I am fixing it

**Review by randallfrank — APPROVED** [review]

**Pull request merged**